dataUtility.py

```diff
@@ -26,13 +30,42 @@ def __init__(self, fname):
         # 'Other' genre is mainly composed of other languages
         df = df[df.genre != "Other"]
         print('Done')
+        if(fout is not None and fpickle is not None and fcount is not None):
+            print('Save data frame to csv...', end='')
+            df.to_csv(fout, encoding="utf8")
+            print('Done')
         print('Finding unique values... ', end='')
         self.base = df
         self.genre = df.genre.unique()
         self.artist = df.artist.unique()
         self.year = df.year.unique()
+        self.count_all = Counter(" ".join(df.lyrics.values.tolist()).split(" "))
         #self.lyrics_vector = df.lyrics.str.split(' ')
         print('Done')
+        if(fpickle is not None and fcount is not None):
+            print('Save data to pickle...', end='')
+            f = open(fpickle, 'wb')
+            fc = open(fcount, 'wb')
+            pickle.dump([" ".join(str(v) for v in self.genre), " ".join(str(v) for v in self.artist), " ".join(str(v) for v in self.year)], f)
```

Comment:
Why are these saved as strings?

Reply:
I saw that pickle can't handle them very well as a list of lists, it could handle them fine as a single list and I didn't want to add too many pickle files for such small sets. I could be wrong though

Reply:
I wonder if you could just pickle self? Might be too much for it to handle though, cPickle might be worth a shot too

Reply:
apparently I can just pickle self, I'l just change it now, if you'd like just close this pull request